pkg/dynamicconfig/reflectors.go

```diff
@@ -66,3 +156,64 @@ func configMapReflector(name string, cs k8sClient.Clientset, t statedb.RWTable[D
 		},
 	}
 }
+
+func ciliumNodeReflector(name string, cs k8sClient.Clientset, t statedb.RWTable[DynamicConfig], l *slog.Logger, index int, sourceLen int, overrides resolver.ConfigOverride) k8s.ReflectorConfig[DynamicConfig] {
+	return k8s.ReflectorConfig[DynamicConfig]{
+		Name:  "node-" + name,
+		Table: t,
+		TransformMany: func(o any) []DynamicConfig {
+			var entries []DynamicConfig
+			node := o.(*corev1.Node).DeepCopy()
+
+			for k, v := range parseNodeConfig(node, l) {
+				priority := getPriorityForKey(k, overrides, index, sourceLen)
+				dc := DynamicConfig{Key: Key{Name: k, Source: node.Name}, Value: v, Priority: priority}
+				entries = append(entries, dc)
+			}
+			return entries
+		},
+		ListerWatcher: utils.ListerWatcherWithModifiers(
+			utils.ListerWatcherFromTyped[*corev1.NodeList](cs.Slim().CoreV1().Nodes()),
+			func(opts *metav1.ListOptions) {
+				opts.FieldSelector = fields.ParseSelectorOrDie(metadataName + name).String()
+			},
+		),
+		QueryAll: func(txn statedb.ReadTxn, t statedb.Table[DynamicConfig]) iter.Seq2[DynamicConfig, statedb.Revision] {
+			return statedb.Filter(
+				t.All(txn),
+				func(dc DynamicConfig) bool {
+					return dc.Key.Source == name
+				},
+			)
+		},
+	}
+}
+
+// parseNodeConfig returns a map of overridable fields from the Node object
+// It allows annotation or labels with config.cilium.io/K=V
+func parseNodeConfig(node *corev1.Node, logger *slog.Logger) map[string]string {
+	out := map[string]string{}
+	read := func(in map[string]string) {
+		for k, v := range in {
+			if !strings.HasPrefix(k, annotation.ConfigPrefix) {
+				continue
+			}
+
+			s := strings.SplitN(k, "/", 2)
+			if len(s) != 2 {
+				logger.Warn("node annotation format invalid", logfields.ConfigAnnotation, k)
+				continue
+			}
+			key := s[1]
+			if errs := apivalidation.IsConfigMapKey(key); len(errs) > 0 {
+				logger.Warn("node annotation format invalid: invalid key", logfields.ConfigAnnotation, k, logfields.Error, errs)
+				continue
+			}
+			out[key] = v
+		}
+	}
+
+	read(node.Labels)
+	read(node.Annotations)
```

Comment:
Ah, I see we are passing in different parts of the node, so the log msgs above when an error occurs should mention which part of the node object failed.

Reply:
we'll log the failing key, so we'll be able to check and I think we also do not officially support this as there are no docs for this but it's a legacy feature supported in config builder. The preferred way is to use CNCs.